src/main/java/com/google/api/codegen/util/CommonAcronyms.java

```diff
@@ -15,44 +15,165 @@
 
 package com.google.api.codegen.util;
 
-import com.google.common.collect.ImmutableMap;
+import com.google.common.base.CaseFormat;
+import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableSet;
 
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 
 /**
  * Utility class to replace fully capitalized common acronyms with an upper camel interpretation.
  */
 public class CommonAcronyms {
-  private static final ImmutableMap<String, String> ACRONYMS =
-      ImmutableMap.<String, String>builder()
-          .put("IAM", "Iam")
-          .put("HTTP", "Http")
-          .put("XML", "Xml")
-          .put("API", "Api")
-          .build();
+  private static final ImmutableSet<String> ACRONYMS =
+      ImmutableSet.<String>builder().add("IAM").add("HTTP").add("XML").add("API").build();
```

Comment:
I don't think so - the point of my implementation here is to actually vary the capitalization of the output. If we need to special-case something like that, it will have to be handled when we get there.